cmds/cpio/cpio.go

```diff
@@ -64,7 +73,38 @@ func main() {
 		}
 
 	case "o":
-		log.Fatalf("no extract yet")
+		var w RecWriter
+		if w, err = NewcWriter(os.Stdout); err != nil {
+			log.Fatalf("%v", err)
+		}
+
+		b := bufio.NewReader(os.Stdin)
+
+		for {
+			var name string
+			if name, err = b.ReadString('\n'); err != nil {
+				if err == io.EOF {
+					err = nil
+				}
+				break
+			}
+			name = strings.TrimRight(name, "\r\n")
+			fi, err := os.Lstat(name)
+			if err != nil {
+				break
+			}
+			f, err := fiToFile(name, fi)
+			if err != nil {
+				break
+			}
+			err = w.RecWrite(f)
+			if err != nil {
+				break
+			}
+		}
+		if err == nil {
+			err = w.RecWrite(trec)
+		}
```

Comment:
I have yet to find one ... this is the case of 'no more errors, write final record'